database.go

```diff
@@ -61,18 +56,27 @@ func NewStateDatabase(ctx context.Context, conf Config) (*stateDatabase, error)
 	}, nil
 }
 
-// ContractCode satisfies Database, it returns the contract code for a give codehash
+// NewStateDatabase returns a new Database implementation using the passed parameters
+func NewStateDatabase(ctx context.Context, conf Config) (*stateDatabase, error) {
+	pgDb, err := NewPGXPool(ctx, conf)
+	if err != nil {
+		return nil, err
+	}
+	return NewStateDatabaseWithPool(pgDb)
+}
+
+// ContractCode satisfies Database, it returns the contract code for a given codehash
 func (sd *stateDatabase) ContractCode(_, codeHash common.Hash) ([]byte, error) {
 	if code := sd.codeCache.Get(nil, codeHash.Bytes()); len(code) > 0 {
 		return code, nil
 	}
-	c, err := keccak256ToCid(ipld.RawBinary, codeHash.Bytes())
+	c, err := util.Keccak256ToCid(ipld.RawBinary, codeHash.Bytes())
 	if err != nil {
 		return nil, fmt.Errorf("cannot derive CID from provided codehash: %s", err.Error())
 	}
 	code := make([]byte, 0)
 	if err := sd.pgdb.QueryRow(context.Background(), GetContractCodePgStr, c).Scan(&code); err != nil {
-		return nil, errNotFound
+		return nil, err
```

Comment:
I had added `errNotFound` cause it looked like the error that would be expected to be returned from geth, but I'm not sure if it would matter in this context (but for example in ipld-eth-server we make sure that the RPC endpoints return the exact error as geth to match the exact same API).

Reply:
Ah, I changed this mainly to help with debugging. If we need it for the API we can add it back or maybe return `not found` from a higher level.